syncer/main.py

```diff
@@ -54,137 +127,369 @@ def __init__(self, webinterface, connector):
         self._config = self.read_config(Syncer.default_config)
         self._config_checker = PluginConfigChecker(Syncer.config_description)
 
-        self._token = None
+        self._polling_interval = 60
+        self._gateways = {}
+        self._name = ''
+        self._output_mapping = {}
+        self._shutter_mapping = {}
         self._enabled = False
-        self._previous_outputs = set()
-        self._read_config()
+        thread = Thread(target=self._read_config)
+        thread.start()
+
+        self.connector.output.subscribe_status_event(handler=self.handle_output_status, version=2)
+        self.connector.shutter.subscribe_status_event(handler=self.handle_shutter_status, version=2)
 
         logger.info("Started Syncer plugin")
 
     def _read_config(self):
```

Comment:
in general, split this function up in separate parts, this is a hard to read part in one go, you can split it up in sensors/outputs/shutters I guess

Reply:
Yeah I know, but I don't feel like doing extra effort haha, maybe if I have more time